cli/src/migrations/schema.ts

```diff
@@ -48,16 +48,22 @@ export const migrationFile = z.object({
   operations: z.array(migrationOperation)
 }) satisfies z.ZodType<Schemas.MigrationObject>;
 
+export const pgRollMigrationHistoryObject = z.object({
+  name: z.string(),
+  operations: z.array(z.any())
+});
+
 export const pgRollMigrationsFile = z.object({
   name: z.string(),
-  migration: z.string(),
+  migration: pgRollMigrationHistoryObject,
   startedAt: z.string(),
   parent: z.string().optional(),
   done: z.boolean(),
   migrationType: z.enum(['pgroll', 'inferred']) satisfies z.ZodType<Schemas.PgRollMigrationType>
-}) satisfies z.ZodType<Schemas.PgRollMigrationHistoryItem>;
+}) satisfies z.ZodType<LocalMigrationFilePgroll>;
 
-export const pgRollMigrationHistoryObject = z.object({
-  name: z.string(),
-  operations: z.array(z.any())
-});
+export type LocalMigrationFilePgroll = {
+  [K in keyof Schemas.PgRollMigrationHistoryItem]: K extends 'migration'
+    ? z.infer<typeof pgRollMigrationHistoryObject>
+    : Schemas.PgRollMigrationHistoryItem[K];
+};
```

Comment:
Nice work with a key remapping type!! Love to see them in the wild 🚀 

In this case, however, maybe we can simplify it though so it's more readable?

Haven't tried them in a real IDE but a couple of alternative suggestions:

```suggestion
export type LocalMigrationFilePgroll = Schemas.PgRollMigrationHistoryItem & {
  migration: z.infer<typeof pgRollMigrationHistoryObject>
};
```

```suggestion
export type LocalMigrationFilePgroll = Omit<Schemas.PgRollMigrationHistoryItem, 'migration'> & {
  migration: z.infer<typeof pgRollMigrationHistoryObject>
};
```

If you think the key remapped type is clear enough we can leave it as-is, up to you 😉